addons/sale_timesheet/static/src/components/so_line_field/so_line_field.js

```diff
@@ -11,7 +11,10 @@ export class SoLineField extends Many2OneField {
         const update = this.update;
         this.update = (value, params = {}) => {
             update(value, params);
-            if (value || this.updateOnEmpty) {
+            if ( // field is unset AND the old & new so_lines are different
+                !this.props.record.data.is_so_line_edited &&
+                this.props.value[0] != (value && value[0].id)
```

Comment:
wasn't the default value of this.updateOnEmpty true (for Many2OneField)? So if the updateOnEmpty is removed shouldn't we just remove the if and do the `this.props.record.update({ is_so_line_edited: true });` in any case?

Reply:
For `Many2OneField` yes, for `ListMany2OneField` it was `false`. The bug occurs in list view, but it's hard to say if the `false` default was intended here, as this conditional was added after `updateOnEmpty` had already been removed from `Many2OneField`, plus there was some other pre-OWL code back then that set `is_so_field_edited` to `True` when clearing the field in list view.
So I contacted @xavierbol who made the change, and he suggested a conditional like this (checking if the field isn't `true` already and whether the update value us different).

Reply:
Hello @nle-odoo 

Thanks for your review! 🙂 

We could indeed just remove the if and do `this.props.record.update({ is_so_line_edited: true });` but I think it would be clearer to also check if it is really needed to do that update (I would have had to do it when I did the conversion). 🙂